### PR TITLE
[pmp-library] add port

### PR DESCRIPTION
**Pull request by cdcseacave**

### CODE REVIEW

**Comment by Adela0814** [comment]
@cdcseacave Please fix CI error and add version file using `vcpkg x-add-version pmp-library`.
https://dev.azure.com/vcpkg/public/_build/results?buildId=93924&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by Adela0814** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/pmp-library/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,25 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO pmp-library/pmp-library
+    REF 3.0.0
```
Comment (dg0yt) [inline ports/pmp-library/portfile.cmake:4]:
```suggestion
    REF "${VERSION}"
```

File: ports/pmp-library/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        -DPMP_BUILD_EXAMPLES=OFF
+        -DPMP_BUILD_TESTS=OFF
+        -DPMP_BUILD_DOCS=OFF
+        -DPMP_BUILD_VIS=OFF
+        -DPMP_STRICT_COMPILATION=OFF
+)
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/pmp")
```
Comment (dg0yt) [inline ports/pmp-library/portfile.cmake:21]: Just guessing now:

```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/pmp" PACKAGE_NAME pmp)
```

File: ports/pmp-library/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "pmp-library",
+  "version": "3.0.0",
+  "port-version": 2,
```
Comment (dg0yt) [inline ports/pmp-library/vcpkg.json:4]: The first time a version is added, the port version is null, and the line is to be removed.
```suggestion
```

File: versions/p-/pmp-library.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    {
+      "git-tree": "f7a3d53185b6dacdd48d64908161de7923ada4df",
+      "version": "3.0.0",
+      "port-version": 0
+    },
+    {
+      "git-tree": "ce4b3416a24a21a44a6c49461b1bcd8290cd8630",
+      "version": "3.0.0",
+      "port-version": 2
+    }
```
Comment (Adela0814) [inline versions/p-/pmp-library.json:12]:
```suggestion
```

Please make sure only one version is added to each modified port's versions file.

File: ports/pmp-library/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        -DPMP_STRICT_COMPILATION=OFF
+)
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/pmp" PACKAGE_NAME pmp)
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+
+file(INSTALL "${SOURCE_PATH}/LICENSE.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (Adela0814) [inline ports/pmp-library/portfile.cmake:25]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE.txt")
```

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by cdcseacave** [comment]
@Adela0814 can  you help pls with merging this?

**Comment by Adela0814** [comment]
@cdcseacave  Please fix the CI error first. please revert to "ready for review" if this PR ready to review again.

**Comment by cdcseacave** [comment]
I'd gladly fix the errors if there would be any info what the problem is; this is all I see:
image
what I am supposed to understand from this message?
pls hlep fixing these cryptic errors

**Comment by cdcseacave** [comment]
and how can I set the label ready for review if I am not allowed to set labels?

**Comment by Adela0814** [comment]
> I'd gladly fix the errors if there would be any info what the problem is; this is all I see: image what I am supposed to understand from this message? pls hlep fixing these cryptic errors

You should rerun `vcpkg x-add-version pmp-library` to update the version file.

**Comment by Adela0814** [comment]
> and how can I set the label ready for review if I am not allowed to set labels?

You can click this button to active your PR.
image

**Review by Adela0814 — APPROVED** [review]

**Pull request merged**